pkg/controllers/management/authprovisioningv2/authprovisioningv2.go

```diff
@@ -86,6 +90,9 @@ func Register(ctx context.Context, clients *wrangler.Context) error {
 	clients.Mgmt.ProjectRoleTemplateBinding().OnChange(ctx, "auth-prov-v2-prtb", h.OnPRTB)
 	clients.Provisioning.Cluster().OnChange(ctx, "auth-prov-v2-cluster", h.OnCluster)
 	clients.CRD.CustomResourceDefinition().OnChange(ctx, "auth-prov-v2-crd", h.OnCRD)
+	if features.RKE2.Enabled() {
+		clients.Dynamic.OnChange(ctx, "auth-prov-v2-rke-machine-config", validMachineConfigGVK, h.OnMachineConfigChange)
```

Comment:
How are you calling validMachineConfigGVK without the schema?

Reply:
In this case, `validMachineConfigGVK` is a function passed to the `OnChange` method. The `OnChange` method will use `validMachineConfigGVK` to determine if this controller wants to look at particular objects.